projects/project.cpp

```diff
@@ -512,92 +512,92 @@ namespace projects {
    }
 
    /*
-     Refine cells of mpiGrid. Each project that wants refinement shoudl implement this function. 
-     Base class function prints a warning and does nothing.
+     Refine cells of mpiGrid up to 2 levels in a box region. This can be overrided in each child project.
     */
-   bool Project::refineSpatialCells( dccrg::Dccrg<spatial_cell::SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid ) const {
+   bool Project::refineSpatialCells(dccrg::Dccrg<spatial_cell::SpatialCell, dccrg::Cartesian_Geometry>& mpiGrid) const {
       int myRank;
-      MPI_Comm_rank(MPI_COMM_WORLD,&myRank);
+      MPI_Comm_rank(MPI_COMM_WORLD, &myRank);
       if (myRank == MASTER_RANK) {
-         cerr << "(Project.cpp) Base class 'refineSpatialCells' in " << __FILE__ << ":" << __LINE__ << " called. Make sure that this is correct." << endl;
+         cerr << "(Project.cpp) Base class 'refineSpatialCells' in " << __FILE__ << ":" << __LINE__
+              << " called. Make sure that this is correct." << endl;
       }
-      
-      MPI_Comm_rank(MPI_COMM_WORLD,&myRank);
-      
-      if(myRank == MASTER_RANK) std::cout << "Maximum refinement level is " << mpiGrid.mapping.get_maximum_refinement_level() << std::endl;
-      
+
+      MPI_Comm_rank(MPI_COMM_WORLD, &myRank);
+
+      if (myRank == MASTER_RANK)
+         std::cout << "Maximum refinement level is " << mpiGrid.mapping.get_maximum_refinement_level() << std::endl;
+
       std::vector<bool> refineSuccess;
-      
+
       for (uint i = 0; i < 2 * P::amrBoxHalfWidthX; ++i) {
          for (uint j = 0; j < 2 * P::amrBoxHalfWidthY; ++j) {
             for (uint k = 0; k < 2 * P::amrBoxHalfWidthZ; ++k) {
-               
-               std::array<double,3> xyz;
+               std::array<double, 3> xyz;
                xyz[0] = P::amrBoxCenterX + (0.5 + i - P::amrBoxHalfWidthX) * P::dx_ini;
                xyz[1] = P::amrBoxCenterY + (0.5 + j - P::amrBoxHalfWidthY) * P::dy_ini;
                xyz[2] = P::amrBoxCenterZ + (0.5 + k - P::amrBoxHalfWidthZ) * P::dz_ini;
-               
+
                CellID myCell = mpiGrid.get_existing_cell(xyz);
+#ifndef NDEBUG
```

Comment:
Umm, this is now incorrect - we do want it to refine even if debug mode isn't on! Same for L575.

Reply:
Yes right. I misunderstood the method `refine_completely_at()` in the if statement.